lightning/src/ln/onion_route_tests.rs

```diff
@@ -240,17 +241,54 @@ impl Writeable for BogusOnionHopData {
 	}
 }
 
+const BADONION: u16 = 0x8000;
+const PERM: u16 = 0x4000;
+const NODE: u16 = 0x2000;
+const UPDATE: u16 = 0x1000;
+
 #[test]
-fn test_onion_failure() {
-	use ln::msgs::ChannelUpdate;
-	use ln::channelmanager::CLTV_FAR_FAR_AWAY;
-	use bitcoin::secp256k1;
+fn test_fee_failures() {
+	// When this test was written, the default base fee floated based on the HTLC count.
+	// It is now fixed, so we simply set the fee to the expected value here.
+	let mut config = test_default_channel_config();
+	config.channel_options.fee_base_msat = 196;
+
+	let chanmon_cfgs = create_chanmon_cfgs(3);
+	let node_cfgs = create_node_cfgs(3, &chanmon_cfgs);
+	let node_chanmgrs = create_node_chanmgrs(3, &node_cfgs, &[Some(config), Some(config), Some(config)]);
+	let mut nodes = create_network(3, &node_cfgs, &node_chanmgrs);
+	let channels = [create_announced_chan_between_nodes(&nodes, 0, 1, InitFeatures::known(), InitFeatures::known()), create_announced_chan_between_nodes(&nodes, 1, 2, InitFeatures::known(), InitFeatures::known())];
+	let logger = test_utils::TestLogger::new();
+	let route = get_route(&nodes[0].node.get_our_node_id(), &nodes[0].net_graph_msg_handler.network_graph.read().unwrap(), &nodes[2].node.get_our_node_id(), Some(InvoiceFeatures::known()), None, &Vec::new(), 40_000, TEST_FINAL_CLTV, &logger).unwrap();
+
+	// positive case
+	let (payment_preimage_success, payment_hash_success, payment_secret_success) = get_payment_preimage_hash!(nodes[2]);
+	nodes[0].node.send_payment(&route, payment_hash_success, &Some(payment_secret_success)).unwrap();
+	check_added_monitors!(nodes[0], 1);
+	pass_along_route(&nodes[0], &[&[&nodes[1], &nodes[2]]], 40_000, payment_hash_success, payment_secret_success);
+	claim_payment(&nodes[0], &[&nodes[1], &nodes[2]], payment_preimage_success);
+
+	let (_, payment_hash, payment_secret) = get_payment_preimage_hash!(nodes[2]);
+	run_onion_failure_test("fee_insufficient", 0, &nodes, &route, &payment_hash, &payment_secret, |msg| {
+		msg.amount_msat -= 1;
+	}, || {}, true, Some(UPDATE|12), Some(msgs::HTLCFailChannelUpdate::ChannelClosed { short_channel_id: channels[0].0.contents.short_channel_id, is_permanent: true}));
 
-	const BADONION: u16 = 0x8000;
-	const PERM: u16 = 0x4000;
-	const NODE: u16 = 0x2000;
-	const UPDATE: u16 = 0x1000;
+	// In an earlier version, we spuriously failed to forward payments if the expected feerate
+	// changed between the channel open and the payment.
```

Comment:
Is this a reference to the `msat-vs-sat bug` mentioned in the commit msg? Maybe missing something, but where can I find more information about the `msat-vs-sat bug`?

Reply:
No, literally current code is "we decide what we think the fee should be when we receive a payment to forward, and dont bother to update our announcements".....

Reply:
Oops, sorry, an earlier version of this patchset had separated out the bugfixes for the two bugs, but now they're in the same, I've updated the commit message to be more clear.